File: app/src/main/java/com/itachi1706/busarrivalsg/BusArrayJSON.java
package com.itachi1706.busarrivalsg;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;

import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Created by Kenneth on 17/6/2015
 * for SingBuses in package com.itachi1706.busarrivalsg
 */
public class BusArrayJSON {

    private BusJSON[] storage;

    public BusJSON[] getStorage() {
        return storage;
    }

    @Override
    public String toString() {
        return "BusArrayObject{" +
                "storage=" + storage + "}";
    }
}

class BusJSON {
    private JsonObject service, operator, stop;

    JsonObject getService() {
        return service;
    }

    JsonObject getOperator() {
        return operator;
    }

    JsonObject getStop() {
        return stop;
    }

    @Override
    public String toString() {
        return "BusObject{" +
                "service=" + service + ", operator=" + operator + "stop=" + stop + "}";
    }
}

File: app/src/main/java/com/itachi1706/busarrivalsg/gsonObjects/offline/BusArrayJSON.kt
package com.itachi1706.busarrivalsg.gsonObjects.offline

/**
 * Created by Kenneth on 17/6/2015
 * for SingBuses in package com.itachi1706.busarrivalsg.gsonObjects.offline
 */
data class BusArrayJSON(val storage: Array<BusJSON>? = null) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as BusArrayJSON

        if (storage != null) {
            if (other.storage == null) return false
            if (!storage.contentEquals(other.storage)) return false
        } else if (other.storage != null) return false

        return true
    }

    override fun hashCode(): Int {
        return storage?.contentHashCode() ?: 0
    }
}
